float64 x_pos
float64 y_pos
